packages/smooth_app/lib/background/background_task_full_refresh.dart

```diff
@@ -75,32 +85,26 @@ class BackgroundTaskFullRefresh extends BackgroundTask {
         productsWithKP.add(barcode);
       }
     }
-    if (productsWithoutKP.isNotEmpty) {
-      final List<ProductField> fieldsWithoutKP = List<ProductField>.from(
-        ProductQuery.fields,
-        growable: true,
-      );
-      fieldsWithoutKP.remove(ProductField.KNOWLEDGE_PANELS);
-
-      await _loadProducts(
-        barcodes: productsWithoutKP,
-        fields: fieldsWithoutKP,
-        daoProduct: daoProduct,
-      );
-    }
-
-    if (productsWithKP.isNotEmpty) {
-      await _loadProducts(
-        barcodes: productsWithKP,
-        fields: ProductQuery.fields,
-        daoProduct: daoProduct,
-      );
-    }
+    await _startDownloadTask(
+      barcodes: productsWithoutKP,
+      work: BackgroundTaskProgressing.workFreshWithoutKP,
+      localDatabase: localDatabase,
+      downloadFlag: BackgroundTaskDownloadProducts.flagMaskExcludeKP,
```

Comment:
Not sure but shouldn't it be downloadFlag

Reply:
> Not sure but shouldn't it be downloadFlag

Here there's no such thing as an initial `downloadFlag`.
But then we start two download tasks: one with a `downloadFlag` that excludes the knowledge panels, and another one with all the product data.